base/compiler/abstractinterpretation.jl

```diff
@@ -547,7 +547,13 @@ function collect_slot_refinements(𝕃ᵢ::AbstractLattice, applicable::Vector{A
             sigt = Bottom
             for j = 1:length(applicable)
                 match = applicable[j]::MethodMatch
```

Comment:
The common way it gets dealt with elsewhere is to add a check like this using this predicate:
```suggestion
                match = applicable[j]::MethodMatch
                valid_as_lattice(match.spec_types, true) || continue
```

Reply:
Ok, so it's valid to assume that if `valid_as_lattice(match.spec_types, true)` is false, that match can never be called? I wasn't sure of that initially which is why I widened to Any, but probably makes sense

Reply:
Right, that is the purpose of that predicate (and in particular, to prevent something like Type{1} from getting into inference somewhere it assumes there will be a valid type, like here)